### PR TITLE
ARF report server

### PR DESCRIPTION
**Pull request by JAORMX**
This is a small server that will listen for requests from all log
collectors and store the results in a PVC. this way, the results won't
take as much space as they used to.

### CODE REVIEW

File: cmd/resultserver/main.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+	if err := srvCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func server(c *config) {
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		filename := r.Header.Get("X-Report-Name")
```
Comment (mrogers950) [inline cmd/resultserver/main.go]: @JAORMX we might want to make this server handle both the pod payloads and the view requests. The pod payloads could be under a subpath, and we could even gate that separately with oauth-proxy if needed. Then we will only need one Service. 
The alternative to that is to keep the receiver and server services seperately, in which this part of the code should be named something like "result-receiver"

File: cmd/resultscollector/main.go [hunk 2]
```diff
@@ -245,6 +255,39 @@ func main() {
...
 				os.Exit(1)
 			}
+			fmt.Println("Uploaded ConfigMap")
+
+			err = backoff.Retry(func() error {
+				url := scapresultsconf.ResultServerURI
+				fmt.Printf("Trying to upload to resultserver: %s\n", url)
+				reader := bytes.NewReader(contents)
+				client := &http.Client{}
+				req, err := http.NewRequest("POST", url, reader)
```
Comment (jhrozek) [inline cmd/resultscollector/main.go]: Does it make sense to check the error here?

File: cmd/resultserver/main.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	"os"
+	"path"
+
+	"github.com/spf13/cobra"
+)
+
+func defineFlags(cmd *cobra.Command) {
+	cmd.Flags().String("address", "127.0.0.1", "Server address")
+	cmd.Flags().String("port", "443", "Server port")
+	cmd.Flags().String("volume", "", "Volume name ??")
```
Comment (jhrozek) [inline cmd/resultserver/main.go]: Is this currently unused?

File: cmd/resultscollector/main.go (hunk 2 above)
Comment (JAORMX) [reply cmd/resultscollector/main.go]: uhm... perhaps not. The error would only be returned if we gave the wrong parameters.

File: cmd/resultserver/main.go (hunk 3 above)
Comment (JAORMX) [reply cmd/resultserver/main.go]: yep, leftover from the original implementation.

File: cmd/resultscollector/main.go (hunk 2 above)
Comment (JAORMX) [reply cmd/resultscollector/main.go]: Fixed.

File: cmd/resultserver/main.go (hunk 3 above)
Comment (JAORMX) [reply cmd/resultserver/main.go]: Fixed.

File: cmd/resultserver/main.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+		}
+		encoding := r.Header.Get("Content-Encoding")
+		extraExtension := encoding
+		if encoding != "" && encoding != "bzip2" {
+			log.Println("Rejecting. Invalid \"Content-Encoding\" header given.")
+			http.Error(w, "invalid content encoding header", 400)
+			return
+		} else if encoding == "bzip2" {
+			extraExtension = "." + extraExtension
+		}
```
[the remark this answers is not in the document]
Comment (JAORMX) [reply cmd/resultserver/main.go]: The thing is, that an empty string is also valid. So I would need three if statements here to handle that.

File: deploy/operator.yaml [hunk 5]
```diff
@@ -34,6 +34,10 @@ spec:
               value: "compliance-operator"
             - name: OPENSCAP_IMAGE
               value: "quay.io/jhrozek/openscap-ocp:remediations_demo"
+            - name: LOG_COLLECTOR_IMAGE
+              value: "quay.io/compliance-operator/resultscollector:latest"
+            - name: RESULT_SERVER_IMAGE
+              value: "quay.io/compliance-operator/resultserver:latest"
```
[the remark this answers is not in the document]
Comment (JAORMX) [reply deploy/operator.yaml]: No, @mrogers950 was looking at that.

**Comment by JAORMX** [comment]
retriggered CI, bootstrapping failed.

**Pull request merged**